### PR TITLE
Update DG: Find

### PR DESCRIPTION
**Pull request by lyeyixian**
HALPPPP!!

### CODE REVIEW

**Comment by lyeyixian** [comment]
I think the commit is messed up, cos I branch this from the branch where I updated the Design section. So, this PR includes both changes. RIP

**Review by jiaweiteo — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -117,22 +117,51 @@ The `Model`,
...
+#### Implementation
+
+Current implementation of Find feature is facilitated by `NameContainsKeywordsPredicate` which implements 
+`Predicate<Nameable>` and has the **test** method overriden with a custom implementations.
+ 
+The `NameContainsKeywordsPredicate#test(Nameable)` is as follows:  
+* loops through the **keywords** (of type `List<String>`) and return true if there is any match with the `Nameable`
+
+`Nameable` provides the following operation:
+* `Nameable#getName()` - Return the name attribute of a `TravelPlanObject`
```
Comment (jeannetoh99) [inline docs/DeveloperGuide.md:147]: Returns instead of Return? And not sure if there should be a full stop here.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -117,22 +117,51 @@ The `Model`,
...
 
+### Find Feature
+
+#### Implementation
+
+Current implementation of Find feature is facilitated by `NameContainsKeywordsPredicate` which implements 
+`Predicate<Nameable>` and has the **test** method overriden with a custom implementations.
+ 
+The `NameContainsKeywordsPredicate#test(Nameable)` is as follows:  
+* loops through the **keywords** (of type `List<String>`) and return true if there is any match with the `Nameable`
```
Comment (jeannetoh99) [inline docs/DeveloperGuide.md:144]: Not sure if there should be a full stop here.

Comment (jeannetoh99) [reply docs/DeveloperGuide.md:144]: Should it be:
* loops through the **keywords** (of type `List<String>`) and return true if there is any match with `Nameable.getName().name`

**Review by jeannetoh99 — APPROVED** [review]
LGTM. Comments on minor stuffs. Can ignore if you think they are okay!

**Pull request merged**